lib/workers/repository/errors-warnings.ts

```diff
@@ -3,81 +3,86 @@ import type { RenovateConfig } from '../../config/types';
 import { logger } from '../../logger';
 import type { PackageFile } from '../../modules/manager/types';
 import { emojify } from '../../util/emoji';
+import type { DepWarnings } from '../types';
 
 export function getWarnings(config: RenovateConfig): string {
-  const warnings = config.warnings!;
-  if (!warnings.length) {
+  if (!config.warnings?.length) {
     return '';
   }
-  let warningText = `\n# Warnings (${warnings.length})\n\n`;
+  let warningText = `\n# Warnings (${config.warnings.length})\n\n`;
   warningText += `Please correct - or verify that you can safely ignore - these warnings before you merge this PR.\n\n`;
-  warnings.forEach((w) => {
+  for (const w of config.warnings) {
     warningText += `-   \`${w.topic}\`: ${w.message}\n`;
-  });
+  }
   warningText += '\n---\n';
   return warningText;
 }
 
 export function getErrors(config: RenovateConfig): string {
-  let errorText = '';
-  const errors = config.errors!;
-  if (!errors.length) {
+  if (!config.errors?.length) {
     return '';
   }
-  errorText = `\n# Errors (${errors.length})\n\n`;
+  let errorText = `\n# Errors (${config.errors.length})\n\n`;
   errorText += `Renovate has found errors that you should fix (in this branch) before finishing this PR.\n\n`;
-  errors.forEach((e) => {
+  for (const e of config.errors) {
     errorText += `-   \`${e.topic}\`: ${e.message}\n`;
-  });
+  }
   errorText += '\n---\n';
   return errorText;
 }
 
-export function getDepWarnings(
+function getDepWarnings(
   packageFiles: Record<string, PackageFile[]>
-): string {
-  let warningText = '';
-  try {
-    const warnings: string[] = [];
-    const warningFiles: string[] = [];
-    for (const files of Object.values(packageFiles || {})) {
-      for (const file of files || []) {
-        if (file.deps) {
-          for (const dep of file.deps || []) {
-            if (dep.warnings?.length) {
-              const message = dep.warnings[0].message;
+): DepWarnings {
+  const warnings: string[] = [];
+  const warningFiles: string[] = [];
+  for (const files of Object.values(packageFiles ?? {})) {
+    for (const file of files ?? []) {
+      if (file.deps) {
+        for (const dep of file.deps ?? []) {
+          if (dep.warnings?.length) {
+            for (const w of dep.warnings) {
+              const message = w.message;
               if (!warnings.includes(message)) {
                 warnings.push(message);
               }
-              if (!warningFiles.includes(file.packageFile!)) {
-                warningFiles.push(file.packageFile!);
+              if (
+                file.packageFile &&
+                !warningFiles.includes(file.packageFile)
+              ) {
```

Comment:
```suggestion
              if (!warningFiles.includes(file.packageFile)) {
```

we should use a `Set<string>` instead for perf